mono/metadata/file-mmap-windows.c

```diff
@@ -175,16 +178,22 @@ open_handle (void *handle, const gunichar2 *mapName, gint mapName_length, int mo
 	HANDLE result = NULL;
 
 	if (mode == FILE_MODE_CREATE_NEW || handle != INVALID_HANDLE_VALUE) {
+		MONO_ENTER_GC_SAFE;
 		result = CreateFileMappingW (handle, NULL, get_page_access (access) | options, (DWORD)(((guint64)*capacity) >> 32), (DWORD)*capacity, mapName);
+		MONO_EXIT_GC_SAFE;
 		if (result && GetLastError () == ERROR_ALREADY_EXISTS) {
+			MONO_ENTER_GC_SAFE;
 			CloseHandle (result);
+			MONO_EXIT_GC_SAFE;
 			result = NULL;
 			*ioerror = FILE_ALREADY_EXISTS;
 		} else if (!result && GetLastError () != NO_ERROR) {
 			*ioerror = convert_win32_error (GetLastError (), COULD_NOT_OPEN);
 		}
 	} else if (mode == FILE_MODE_OPEN || mode == FILE_MODE_OPEN_OR_CREATE && access == MMAP_FILE_ACCESS_WRITE) {
+		MONO_ENTER_GC_SAFE;
 		result = OpenFileMappingW (get_file_map_access (access), FALSE, mapName);
```

Comment:
Surely this does no I/O. It opens an in-memory construct. But it'd have to use locks, briefly.